src/common/post.ts

```diff
@@ -1104,3 +1222,53 @@ export const ensurePostAnalyticsPermissions = async ({
     );
   }
 };
+
+export const createFreeformPost = async (
+  ctx: AuthContext,
+  args: CreatePostArgs,
+) => {
```

Comment:
This is the old `createFreeformPost` function plus the checks that the endpoint was doing before calling it